shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1329,9 +1328,8 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool)
       case TypedNuCls(level, td, ttu, tparams, params, members, thisTy) =>
         ectx(tparams) |> { implicit ectx =>
           NuTypeDef(td.kind, td.nme, td.tparams,
-            Tup(params.map(p => N -> Fld(false, false, Asc(p._1, go(p._2.ub))))),
+            S(Tup(params.map(p => N -> Fld(false, false, Asc(p._1, go(p._2.ub)))))),
```

Comment:
The correct thing to do here is actually `Option.when(td.params.isDefined)(Tup(...))`.